topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+|`Producer`
+
+h|Kafka property name
+|`compression.type`
+|===
+--
+
+
+Message format version::
++
+--
+Message format version is the `ApiVersion` value that the broker uses to append messages to topics. This value must be a valid `ApiVersion` value, such as 0.8.2, 0.9.0.0, or 0.10.0.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|String
+
+h|Default value
+|`2.7-IV2`
+
+h|Kafka property name
+|`message.format.version`
+|===
+--
+
+
+=== Log
+
+These properties define how your log is handled.
+
+NOTE: Messages are continually appended to the partition log and are assigned their offset.
+
+Cleanup policy::
++
+--
+Cleanup policy determines whether log messages are deleted, compacted, or both. With the `Compact, Delete` option, log segments are first compacted and then deleted according to the retention time or size limit settings.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|Yes
+
+h|Type
+|List
+
+h|Default value
+|`Delete`
+
+h|Supported values
+|`Delete`, `Compact`, `Compact, Delete`
+
+h|Kafka property name
+|`cleanup.policy`
+|===
+--
+
+Delete retention time::
++
+--
+Delete retention time is the amount of time that deletion tombstone markers are retained if the topic is compacted.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`86400000 ms` (1 day)
+
+h|Kafka property name
+|`delete.retention.ms`
+|===
+--
+
+Minimum cleanable dirty ratio::
++
+--
+Minimum cleanable dirty ratio is the ratio of entries in the log that can be compacted versus entries that cannot be compacted. When this ratio is reached, the eligible messages in the log are compacted. By default, the ratio is `0.5` or 50%, meaning that messages are compacted after at least half of the log messages are eligible. This property applies only when the topic cleanup policy is set to `Compact` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Double
+
+h|Default value
+|`0.5` (50%)
+
+h|Kafka property name
+|`min.cleanable.dirty.ratio`
+|===
+--
+
+Minimum compaction lag time::
++
+--
+Minimum compaction lag time is the minimum time a message remains uncompacted in a log. This property applies only when the topic cleanup policy is set to `Compact` or `Compact, Delete`.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`0 seconds`
+
+h|Kafka property name
+|`min.compaction.lag.ms`
+|===
+--
+
+=== Replication
+
+These properties control the behavior of your replicas. Each of these properties impacts every replica created in the topic.
+
+Unclean leader election::
++
+--
+Unclean leader election allows a follower replica that is not in sync with the partition leader to become the leader of the partition. This property provides a way to retain at least partial data if partition leaders are jeopardized. However, this property can lead to data loss, so it is disabled by default.
```

Comment:
Mention that by disabling this property, clean election is always used? Clean leader election to an in-sync replica guarantees no loss of data.

Reply:
_jeopardized_ might not be clear. Fails or goes offline? Unable to pick up messages?

Reply:
Yes, already mention that it's disabled by default due to the potential for data loss. And agreed on "jeopardized", changed to "... leaders are lost". That work?